app/javascript/Components/Result/marks_panel.jsx

```diff
@@ -13,22 +13,45 @@ export class MarksPanel extends React.Component {
     super(props);
     this.state = {
       expanded: new Set(),
+      activeCriterionId: null,
     };
   }
 
+  setActiveCriterion = id => {
+    this.setState({activeCriterionId: id});
+  };
+
+  nextCriterion = () => {
+    const criteria = this.props.marks.filter(
+      c => !(this.props.assigned_criteria !== null && !this.props.assigned_criteria.includes(c.id))
```

Comment:
The negation here is awkward, you can simplify the condition by pushing the negation in